src/solver/optim/ortoolsImpl/include/antares/solver/optim/ortoolsImpl/linearProblem.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#pragma once
+
+#include <antares/solver/optim/api/linearProblem.h>
+#include <antares/solver/optim/ortoolsImpl/mipConstraint.h>
+#include <antares/solver/optim/ortoolsImpl/mipSolution.h>
+#include <antares/solver/optim/ortoolsImpl/mipVariable.h>
+
+// forward declaration
+namespace operations_research
+{
+class MPSolver;
+class MPSolverParameters;
+class MPObjective;
+} // namespace operations_research
+
+namespace Antares::Solver::Optim::OrtoolsImpl
+{
+
+class OrtoolsLinearProblem final: public Api::ILinearProblem
+{
+public:
+    OrtoolsLinearProblem(bool isMip, const std::string& solverName);
```

Comment:
About : 
```C++
OrtoolsLinearProblem(bool isMip, const std::string& solverName);
```
We have a boolean **isMip** as an argument of the constructor here.
It's used to build either a MIP or a LP problem internally.
On the other hand, we have a list of MIP variables and a list of MIP constraints as data members.
Question :
- If we have a LP problem, isn't that strange and lead to troubles having this LP problem working with MIP variables or constraints ?
- For now, we're about to use class **OrtoolsLinearProblem** only with `isMip == false`. Do we not make things more complex than it should be at this stage ?
- Even if we're sure MILP problem will come into play at some point, should we not make 2 classes instead of one (one for LP, one for MILP), and possible put the common code either in a common parent class or in a polymorphic data member (see pattern strategy) ?

Reply:
> If we have a LP problem, isn't that strange and leads to troubles having this LP problem working with MIP variables or MIP constraints ?

UB or better, error. This is related to how MPSolver handles MIP/LP problems. We might make the decision later in the execution flow (is there any MIP variable ? If so, use MIP, otherwise use LP), at the price of a bit more complexity.

> For now, we're about to use class OrtoolsLinearProblem only with isMip == false. Do we not make things more complex than it should be at this stage ?

No, we'll 100% want to use MIP problems later

> Even if we're sure MILP problem will come into play at some point, should we not make 2 classes instead of one (one for LP, one for MILP), and possibly have the common code either in a common parent class, or in a polymorphic data member (see pattern strategy) ? The goal would be to have smaller classes, reducing one their responsibilities perimeter, as required by software quality.

Duplicated classes wouldn't be easy to use/maintain in my opinion.

Reply:
> Duplicated classes wouldn't be easy to use/maintain in my opinion.

This is not what I suggest. On the contrary, I suggest putting the common code in a separate code unit to avoid duplication. But we'll see that in time.